### PR TITLE
Fix issue #244 - hideIndicator called twice

### PR DESCRIPTION
**Pull request by baumann74**
Fix for issues #244

### CODE REVIEW

**Comment by carsongross** [comment]
Hi There,

Unfortunately this is a situation where we do need to call request cleanup before we snapshot the HTML, to ensure that the DOM is clean before we save it to local storage.  I need to think a bit more about how to best handle this.  :/  

Sorry, I will do so today.

**Comment by baumann74** [comment]
I am running with the above fix and I have not experience any issues. But you are the expert :-)

**Comment by baumann74** [comment]
Hi Carson.
You were right that "we do need to request cleanup before we snapshot". I just realized that my fix did not work when I use ic-push-url="true" on a link and the indicator is inside the element being replace. The spinner is then shown when the "Back" button is pressed because we don't cleanup (remove the indicator) before we take a snapshot for the history. 

I have just submitted a new version of the pull request that seems to work in both situations.  It keeps the call to requestCleanup in on-succes but it goes to the "ic-use-transition" branch in hideIndicator on both on-succes and on-complete. Please have a look.

**Comment by carsongross** [comment]
Thank you for the updated fix, I will review it by the end of the week.

File: src/intercooler.js [hunk 1]
```diff
@@ -72,9 +72,10 @@ var Intercooler = Intercooler || (function() {
   }
 
   function hideIndicator(elt) {
-    if (elt.data('ic-use-transition')) {
+    if (elt.data('ic-use-transition') || elt.data('is-indicator-cleared')) {
```
Comment (jsampson-gw) [inline src/intercooler.js]: Should this be `ic-` instead of `is-`?

File: test/unit_tests.html [hunk 2]
```diff
@@ -2405,7 +2405,66 @@ <h4>Test HTML</h4>
...
+    </a>
+    <i id="ic-push-url-and-ic-use-transition-indicator" class="fa fa-spinner fa-spin ic-use-transition"></i>
+    <div id="test-ic-push-url-and-ic-use-transition-content" ic-history-elt="">
+      <h1>Content</h1>
+    </div>
+  </div>
+  <script>
+    QUnit.test("ic-use-transition should work when using ic-push-url", function (assert) {
+      // don't run test locally
+      if (window.location.hostname == "") {
```
Comment (jsampson-gw) [inline test/unit_tests.html]: This is actually a question for @chg20: I see that this `if` block is copied from the existing test for "Basic history support works", where you added it sometime after originally writing that test. What problem does it prevent? Is it a problem in all browsers?

File: test/unit_tests.html [hunk 3]
```diff
@@ -2405,7 +2405,66 @@ <h4>Test HTML</h4>
         assert.equal($("#test-cancel-with-ic-on-success").text(), "");
       });
   </script>
-
+	
+  <div id="test-ic-push-url-and-ic-use-transition">
```
Comment (jsampson-gw) [inline test/unit_tests.html]: Before or soon after merging, we will need add this test to the other *_unit_tests.html files, either using test/gen_tests.rb or manually pasting these added lines at the bottom of each.

(It looks like there's already one test from an earlier change that is missing from those other files, "Server Sent Event trigger handles missing ic-sse-src element". Whoever runs gen_tests.rb will see that one appear as well as the new one.)

File: src/intercooler.js (hunk 1 above)
Comment (chg20) [reply src/intercooler.js]: For data (rather than attr()) storage, using the ic- prefix is fine IMO.

Comment (baumann74) [reply src/intercooler.js]: Fixed.

File: test/unit_tests.html (hunk 3 above)
Comment (jsampson-gw) [reply test/unit_tests.html]: I'll do this myself after merging this pull request.

File: test/unit_tests.html (hunk 2 above)
Comment (jsampson-gw) [reply test/unit_tests.html]: I tried this myself, and I see now that it does indeed fail when run via `file:///...` without this guard, but it passes when run via `http://localhost/...` (in Chrome).

File: src/intercooler.js (hunk 1 above)
Comment (jsampson-gw) [reply src/intercooler.js]: Thanks.

**Pull request merged**